### PR TITLE
Improved documentation on native dataset support

### PR DESCRIPTION
**Pull request by schlunma**
## Description

The documentation on how to include native datasets is sparse and also wrong at some points. This PR fixes this.

Link to doc:
- How to add support for new native datasets (`develop/fixing_data`): https://esmvaltool--1559.org.readthedocs.build/projects/ESMValCore/en/1559/develop/fixing_data.html#add-support-for-new-native-datasets
- List of supported native datasets (`quickstart/find_data`): https://esmvaltool--1559.org.readthedocs.build/projects/ESMValCore/en/1559/quickstart/find_data.html#datasets-in-native-format
- Example for config-developer entry ( `quickstart/configure`): https://esmvaltool--1559.org.readthedocs.build/projects/ESMValCore/en/1559/quickstart/configure.html#configuring-datasets-in-native-format

## [Checklist](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#checklist-for-pull-requests)

It is the responsibility of the author to make sure the pull request is ready to review. The icons indicate whether the item will be subject to the [🛠 Technical][1] or [🧪 Scientific][2] review.

[1]: https://docs.esmvaltool.org/en/latest/community/review.html#technical-review
[2]: https://docs.esmvaltool.org/en/latest/community/review.html#scientific-review

- [x] [🧪][2] The new functionality is [relevant and scientifically sound](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#scientific-relevance)
- [x] [🛠][1] This pull request has a [descriptive title and labels](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-title-and-label)
- [x] [🛠][1] Code is written according to the [code quality guidelines](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#code-quality)
- [x] [🧪][2] and [🛠][1] [Documentation](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#documentation) is available
- [x] [🛠][1] [Unit tests](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#tests) have been added
- [x] [🛠][1] Changes are [backward compatible](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#backward-compatibility)
- [x] [🛠][1] Any changed [dependencies have been added or removed](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#dependencies) correctly
- [x] [🛠][1] The [list of authors](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#list-of-authors) is up to date
- [x] [🛠][1] All [checks below this pull request](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-checks) were successful

***

To help with the number pull requests:

- 🙏 We kindly ask you to [review](https://docs.esmvaltool.org/en/latest/community/review.html#review-of-pull-requests) two other [open pull requests](https://github.com/ESMValGroup/ESMValCore/pulls) in this repository

### CODE REVIEW

**Review by bouweandela — CHANGES_REQUESTED** [review]
Thanks for looking into this @schlunma! The problem I have with how the documentation is currently organized is that instructions for people who just want to use the data are mixed with the instructions for people who want to add support for a new dataset to the ESMValCore. This pull request, in its current state, seems to make that even worse. Could you have a look if it would be possible to restructure the documentation such that the `quickstart` directory contains documentation for users and the `develop` directory contains documentation for people who want to add support for a new dataset?

**Comment by schlunma** [comment]
Yes, I fully agree with this. I think this actually applies to more parts of `quickstart`, but this is something for other PRs.

I moved the main part on how to add support for new models into `develop/fixing_data`. The `quickstart` section now only includes an overview of the necessary paragraphs in the config-developer file and relevant links.

File: doc/develop/fixing_data.rst [hunk 1]
```diff
@@ -434,6 +454,106 @@ explained in :ref:`extra_facets`, and which content is :download:`available here
...
+Add support for new native datasets
+-----------------------------------
+
+This section describes how to add support for additional native datasets.
+You can choose to host this new data source either under a dedicated project or
+under project ``native6``.
+
+To allow ESMValCore to locate the data files, use the following steps:
+
+   - If you want to use the ``native6`` project (recommended for native
```
Comment (bouweandela) [inline doc/develop/fixing_data.rst:468]: I would say that this is recommended if you can easily move the files to the usual `native6` directory structure, while a custom project is needed if you cannot control the location of the files.

File: doc/develop/fixing_data.rst [hunk 2]
```diff
@@ -329,27 +329,42 @@ strictness to the highest:
   be sure that future users will not be distracted by inoffensive warnings.
 
 
-Natively supported non-CMIP datasets
-====================================
+.. _fixing_native_datasets:
 
-Some fixed datasets and native models formats are supported through
-the ``native6`` project or through a dedicated project.
+Natively supported datasets
```
Comment (bouweandela) [inline doc/develop/fixing_data.rst:334]: Could you move this section, i.e. the list of datasets supported datasets to https://esmvaltool--1559.org.readthedocs.build/projects/ESMValCore/en/1559/quickstart/find_data.html#datasets-in-native-format? I think it does not make sense that people who just want to use some ERA5 data need to dive into the development documentation.

**Review by bouweandela — COMMENTED** [review]
Thanks, look much better already!

File: doc/develop/fixing_data.rst (hunk 2 above)
Comment (schlunma) [reply doc/develop/fixing_data.rst]: Done!

File: doc/quickstart/configure.rst [hunk 3]
```diff
@@ -513,44 +513,41 @@ related to CMOR table settings available:
...
-      such a file for IPSL-CM6
-      <../../esmvalcore/_config/extra_facets/ipslcm-mappings.yml>`.
-
-  - ensuring that ESMValCore get the right metadata and data out of
-    your data files: this is described in :ref:`fixing_data`
+Configuring datasets in native format
+-------------------------------------
+
+ESMValCore can be configured for handling native model output formats and
+specific observation datasets without preliminary reformatting.
```
Comment (bouweandela) [inline doc/quickstart/configure.rst:520]:
```suggestion
specific reanalysis/observation datasets without preliminary reformatting.
```

File: doc/quickstart/configure.rst [hunk 4]
```diff
@@ -513,44 +513,41 @@ related to CMOR table settings available:
...
+These datasets can be either hosted under the ``native6`` project (mostly
+native observational datasets) or under a dedicated project, e.g., ``ICON``
```
Comment (bouweandela) [inline doc/quickstart/configure.rst:522]:
```suggestion
reanalysis/observational datasets) or under a dedicated project, e.g., ``ICON``
```

File: doc/quickstart/find_data.rst [hunk 5]
```diff
@@ -82,6 +75,134 @@ public availability, the ``default`` directory must be structured accordingly
...
 
+
+.. _read_native_datasets:
+
+Datasets in native format
+-------------------------
+
+Some datasets are supported in their native format (i.e., the data is not
+formatted according to a CMIP data request) through the ``native6`` project
+(mostly native observational datasets) or through a dedicated project, e.g.,
```
Comment (bouweandela) [inline doc/quickstart/find_data.rst:86]:
```suggestion
(mostly reanalysis/observational datasets) or through a dedicated project, e.g.,
```

File: doc/quickstart/find_data.rst [hunk 6]
```diff
@@ -82,6 +75,134 @@ public availability, the ``default`` directory must be structured accordingly
...
+``ICON`` (mostly native models).
+A detailed description of how to include new native datasets is given
+:ref:`here <add_new_fix_native_datasets>`.
+
+.. _read_native_obs:
+
+Supported native observational datasets
```
Comment (bouweandela) [inline doc/quickstart/find_data.rst:93]:
```suggestion
Supported reanalysis/observational datasets
```

File: doc/quickstart/find_data.rst [hunk 7]
```diff
@@ -82,6 +75,134 @@ public availability, the ``default`` directory must be structured accordingly
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The following native observational datasets are supported under the ``native6``
```
Comment (bouweandela) [inline doc/quickstart/find_data.rst:96]:
```suggestion
The following reanalysis/observational datasets are supported under the ``native6``
```

**Review by bouweandela — APPROVED** [review]
Thanks, looks good now! Maybe you could consider finding a better name than observational for the supported datasets, because I believe that neither ERA5 nor MSWEP are (purely) observational datasets, but other than that it looks good.

**Comment by schlunma** [comment]
Great, thanks for the review Bouwe!

**Pull request merged**